go/vt/vtgate/vindexes/vschema.go

```diff
@@ -190,15 +203,22 @@ func buildTables(source *vschemapb.SrvVSchema, vschema *VSchema) error {
 				}
 				vindex := vindexes[ind.Name]
 				owned := false
+				exclusive := true
+
+				if vindexesUsageCount[ind.Name] > 1 {
+					exclusive = false
+				}
```

Comment:
I don't think you want to base this solely on usage count -- lots of tables could use the hash vindex and that's irrelevant.

I think the context you want to add here is a `Primary` bit that indicates if this is the primary vindex or not.

Then it seems like the implementation of `canChangeVindexColumns` would only allow columns where Primary is false, type is Lookup, and Owned is true, which _seems_ like the right cases to me.

Then again I'm not 100% confident in my understanding of Owned vs Non-Owned vindexes so @sougou should weigh in here as well.

Reply:
From my understanding when I talked to @sougou  about this feature, we do need to make sure that the vindex is exclusive. 

I agree that setting this value for hash vindexes is irrelevant.  However, I thought this was a simple way to keep track of exclusivity. 

If we find a simpler way to do this, happy to change it.

Reply:
I'm actually changing my mind about this check, for the following reasons:
* We already perform deletes of an owned and shared vindex. So, we are assuming the application is being responsible about deleting rows from the dependent tables first.
* They also have the option of setting up `cascade` deletes, which would auto-cleanup dependent rows: https://dev.mysql.com/doc/refman/5.7/en/create-table-foreign-keys.html.

The above two options are equally applicable for updates. So, why go out of our way to prevent this?